src/main/java/com/team4/testingsystem/controllers/TestsController.java

```diff
@@ -0,0 +1,70 @@
+package com.team4.testingsystem.controllers;
+
+import com.team4.testingsystem.entities.Test;
+
+import com.team4.testingsystem.services.TestsService;
+import com.team4.testingsystem.utils.jwt.JwtTokenUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.DeleteMapping;
+
+@RestController
+@RequestMapping(path="/tests")
+public class TestsController {
+
+    @Autowired
+    private TestsService testsService;
+
+    private long getUserId() {
+
+    return JwtTokenUtil.extractUserDetails().getId();
+    }
+
+    @GetMapping(path = "/")
+    public Iterable<Test> getAll() {
+        return testsService.getAll();
+    }
+
+    @GetMapping(path = "/{id}")
+    public Test getById(@PathVariable("id") long id) {
+        return testsService.getById(id);
+    }
+
+    @PostMapping(path = "/assign/{userId}")
+    public long assign(@PathVariable("userId") long userId) {
+        return testsService.createForUser(userId);
+    }
+
+    @PostMapping(path = "/start")
+    public long startNotAssigned() {
```

Comment:
What test will be generated for user here?

Reply:
The method creates a test for the user, who wants to learn one's level of English without HR's help. The test is created and immediately starts. At the moment it saves only basic information about tests (such as createdAt and startedAt dates (createdAt = startedAt), status = STARTED, id, userId and initial evaluation = 0). All the other things, such as generating of questions, will be added after merging other crud tasks. By the way, the method will have more params (f.e supposed level of Engilsh)

Reply:
Which one? We will have a list of test with different levels